scripts/partition_manager/partition_manager.rst

```diff
@@ -536,11 +536,28 @@ See :ref:`ug_bootloader_external_flash` for more details on using external flash
 
 .. _pm_build_system:
 
-You must enable device drivers for each flash partition.
-Set the ``DEFAULT_DRIVER_KCONFIG`` parameter to the Kconfig option that enables the driver during compilation.
-When a driver for a partition is not enabled, the partition will not be included in the partition manager and will not be accessible at runtime.
-You can use the :kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK` Kconfig option to override the behavior and prevent compile-time check for external flash memory devices.
-This is useful when the external device does not have a driver controlled by Kconfig options.
+For a partition to be accessible at run-time, a flash device it is placed on needs to have its driver enabled at compile time.
+Partition manager will ignore partitions that are located in a region without a driver associated with it.
+To let the partition manager know which Kconfig ensures existence of a driver, the ``DEFAULT_DRIVER_KCONFIG`` option is used.
+Within the :file:`partition_manager.cmake` file the option is assigned the :kconfig:option:`CONFIG_SOC_FLASH_NRF` option for partitions in internal flash, which means that all partitions placed in internal flash are always available at run-time.
+For external regions the :file:`partition_manager.cmake` assigns the ``DEFAULT_DRIVER_KCONFIG`` with :kconfig:option:`CONFIG_PM_EXTERNAL_FLASH_HAS_DRIVER`; the option needs to be set by external driver or Kconfig.
+This option is automatically set when :kconfig:option:`CONFIG_NRF_QSPI_NOR` or :kconfig:option:`CONFIG_SPI_NOR` are enabled, and can be overridden by setting :kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK` in the application configuration.
```

Comment:
```suggestion
For partitions in the internal flash memory, ``DEFAULT_DRIVER_KCONFIG`` within :file:`partition_manager.cmake` is set automatically to :kconfig:option:`CONFIG_SOC_FLASH_NRF`, so that all the partitions placed in the internal flash are always available at runtime.
For external regions, ``DEFAULT_DRIVER_KCONFIG`` within :file:`partition_manager.cmake` must be set to :kconfig:option:`CONFIG_PM_EXTERNAL_FLASH_HAS_DRIVER`.
Out-of-tree drivers can select this value to attest that they provide support for the external flash.
This is an hidden option and can be selected only by an external driver or a Kconfig option.

This option is automatically set when :kconfig:option:`CONFIG_NRF_QSPI_NOR` or :kconfig:option:`CONFIG_SPI_NOR` are enabled.
If the application provides the driver in an unusual way, this option can be overridden by setting :kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK` in the application configuration.
```